nixos/modules/services/home-automation/home-assistant.nix

```diff
@@ -64,6 +65,20 @@ let
     extraComponents = oldArgs.extraComponents or [] ++ extraComponents;
     extraPackages = ps: (oldArgs.extraPackages or (_: []) ps) ++ (cfg.extraPackages ps);
   }));
+
+  # Create a directory that holds all lovelace modules
+  customLovelaceModulesDir = pkgs.buildEnv {
+    name = "home-assistant-custom-lovelace-modules";
+    paths = cfg.customLovelaceModules;
+  };
+
+  # Create parts of the lovelace config that reference lovelave modules as resources
+  customLovelaceModulesResources = {
+    lovelace.resources = map (card: {
+      url = "/local/nixos-lovelace-modules/${card.pname}.js?${card.version}";
```

Comment:
Not sure about the stability of the filename here.

Reply:
For first pass I think it's OK- since edge-cases need to be established first.
You could leave a comment with the concern though

Reply:
Has been addressed. Defaults to `${card.pname}.js`, but can be overridden by setting `passthru.entrypoint`.